tests/unit/cartography/intel/semgrep/test_findings.py

```diff
@@ -0,0 +1,168 @@
+    )
+
+    # Act
+    try:
+        get_deployment("your_semgrep_app_token")
+    except RequestException as e:
+        # Assert
+        assert e.args[0] == "Server Error"
+        assert e.request == mock_requests
+        assert e.response.status_code == 500
+
+
+def mock_get_sca_vulns_response(url, headers, params):
+    mock_response = MagicMock()
+    mock_response.status_code = 200
+    if url == 'https://semgrep.dev/api/sca/deployments/yourorgid/vulns':
+        mock_response.json.return_value = SCA_RESPONSE
+        mock_response.request.url = 'https://semgrep.dev/api/sca/deployments/yourorgid/vulns'
+    else:
+        mock_response.json.return_value = {
+            "vulns": [],
+            "cursor": "789012",
+        }
+        mock_response.request.url = 'https://semgrep.dev/api/sca/deployments/yourorgid/vulns?cursor=123456'
+    return mock_response
+
+
+@patch("cartography.intel.semgrep.findings.requests")
+def test_get_sca_vulns(mock_requests):
+    # Arrange
+
+    mock_requests.get.side_effect = mock_get_sca_vulns_response
+    semgrep_app_token = "your_semgrep_app_token"
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {semgrep_app_token}",
+    }
+
+    # Act
+    vulns = get_sca_vulns(semgrep_app_token, "yourorgid")
+
+    # Assert
+    assert mock_requests.get.any_call(
+        'https://semgrep.dev/api/sca/deployments/yourorgid/vulns',
+        headers=headers,
+    )
+    assert mock_requests.get.any_call(
+        'https://semgrep.dev/api/sca/deployments/yourorgid/vulns',
+        headers=headers, params={'cursor': '123456'},
+    )
+    assert vulns[0]["title"] == "Reachable vuln"
+    assert vulns[0]["repositoryName"] == "yourorg/yourrepo"
+    assert vulns[0]["advisory"]["ruleId"] == "ssc-92af1d99-4fb3-4d4e-a9f4-d57572cd6590"
+    assert vulns[0]["advisory"]["severity"] == "HIGH"
+    assert vulns[0]["advisory"]["references"]["cveIds"][0] == "CVE-2023-37897"
+    assert vulns[0]["advisory"]["reachability"] == "MANUAL_REVIEW_REACHABLE"
+    assert vulns[0]["exposureType"] == "REACHABLE"
+
+
+@patch("cartography.intel.semgrep.findings.requests")
+def test_get_sca_vulns_exception(mock_requests):
+    # Arrange
+    mock_requests.RequestException = RequestException
+    mock_response = MagicMock(status_code=500)
+    mock_requests.get.side_effect = RequestException("Server Error", request=mock_requests, response=mock_response)
+
+    # Act
+    try:
+        get_sca_vulns("your_semgrep_app_token", "yourorgid")
+    except RequestException as e:
+        # Assert
+        assert e.args[0] == "Server Error"
+        assert e.request == mock_requests
+        assert e.response.status_code == 500
+
+
+def test_transform_sca_vulns():
+    # Arrange
+    raw_vulns = SCA_RESPONSE["vulns"]
+    # Act
+    vulns = transform_sca_vulns(raw_vulns)
+    # Assert
+    assert vulns[0]["id"] == "yourorg/yourrepo|ssc-92af1d99-4fb3-4d4e-a9f4-d57572cd6590"
```

Comment:
would be maybe easier to do an equality check against a rendered dict

Reply:
Updated it.